hpat/datatypes/hpat_pandas_seriesgroupby_types.py

```diff
@@ -0,0 +1,123 @@
+# CONTRIBUTORS BE LIABLE FOR ANY DIRECT, INDIRECT, INCIDENTAL, SPECIAL,
+# EXEMPLARY, OR CONSEQUENTIAL DAMAGES (INCLUDING, BUT NOT LIMITED TO,
+# PROCUREMENT OF SUBSTITUTE GOODS OR SERVICES; LOSS OF USE, DATA, OR PROFITS;
+# OR BUSINESS INTERRUPTION) HOWEVER CAUSED AND ON ANY THEORY OF LIABILITY,
+# WHETHER IN CONTRACT, STRICT LIABILITY, OR TORT (INCLUDING NEGLIGENCE OR
+# OTHERWISE) ARISING IN ANY WAY OUT OF THE USE OF THIS SOFTWARE,
+# EVEN IF ADVISED OF THE POSSIBILITY OF SUCH DAMAGE.
+# *****************************************************************************
+
+"""
+
+| :class:`pandas.SeriesGroupBy` type implementation in HPAT
+| Also, it contains related types and iterators for SeriesGroupBy type handling
+
+"""
+
+
+import pandas
+
+from numba import types, cgutils
+from numba.extending import (models, overload, register_model, make_attribute_wrapper, intrinsic)
+from numba.datamodel import (register_default, StructModel)
+from numba.typing.typeof import typeof_impl
+from numba.typing.templates import signature
+
+import hpat
+from hpat.hiframes.pd_series_ext import (
+    SeriesType,
+    is_str_series_typ,
+    if_arr_to_series_type,
+    series_to_array_type,
+    if_series_to_array_type,
+    is_dt64_series_typ)
+
+class SeriesGroupByTypeIterator(types.SimpleIteratorType):
+ 
+    def __init__(self, data):
+        self.data = data
+        super(SeriesGroupByTypeIterator, self).__init__("SeriesGroupByTypeIterator", data)
+ 
+ 
+@register_default(SeriesGroupByTypeIterator)
+class SeriesGroupByTypeIteratorModel(StructModel):
+    def __init__(self, dmm, fe_type):
+        #print("Call @register_default(SeriesGroupByTypeIterator)::ctor with: ", self)
+        members = [
+            ('data', fe_type.data)
+        ]
+        super(SeriesGroupByTypeIteratorModel, self).__init__(dmm, fe_type, members)
+ 
+ 
+make_attribute_wrapper(SeriesGroupByTypeIterator, 'data', '_data')
+
+
+class SeriesGroupByType(types.IterableType):
+    """
+    Type definition for SeriesGroupBy functions handling.
+    """
+
+    def __init__(self, data):
+        self.data = data
+        super(SeriesGroupByType, self).__init__('SeriesGroupByType')
+
+    @property
+    def iterator_type(self):
+        return SeriesGroupByTypeIterator(self)
+
+
+@register_model(SeriesGroupByType)
+class SeriesGroupByModel(StructModel):
+    def __init__(self, dmm, fe_type):
+        members = [
+            ('data', fe_type.data)
+        ]
+        models.StructModel.__init__(self, dmm, fe_type, members)
+
+
+make_attribute_wrapper(SeriesGroupByType, 'data', '_data')
+
+
+@intrinsic
+def init_seriesgroupby(typingctx, data):
+
+    def codegen(context, builder, signature, data_val):
+        series = cgutils.create_struct_proxy(signature.return_type)(context, builder)
+        series.data = data_val
+        
+        if context.enable_nrt:
+            context.nrt.incref(builder, types.Tuple(data), series.data)
```

Comment:
@sklam My previous comment could be addressed by changing `types.Tuple(data)` to `data` here